platform/commonUI/edit/src/actions/SaveAction.js

```diff
@@ -50,23 +67,72 @@ define(
         SaveAction.prototype.perform = function () {
             var domainObject = this.domainObject,
                 $location = this.$location,
-                urlService = this.urlService;
+                urlService = this.urlService,
+                self = this;
+
+            function doWizardSave(domainObject, parent) {
+                var context = domainObject.getCapability("context");
+                var wizard = new CreateWizard(domainObject.useCapability('type'), parent, self.policyService);
+
+                // Create and persist the new object, based on user
+                // input.
+                function persistResult(formValue) {
+                    var parent = wizard.getLocation(formValue),
+                        newModel = wizard.createModel(formValue);
+                    return self.creationService.createObject(newModel, parent);
+                }
+
+                function doNothing() {
+                    // Create cancelled, do nothing
+                    return false;
+                }
+
+                /**
+                 * Add the composees of the 'virtual' object to the
+                 * persisted object
+                 * @param object
+                 * @returns {*}
+                 */
+                function composeObject(object){
+                    return object && self.$q.when(object.hasCapability('composition') && domainObject.hasCapability('composition'))
+                    .then(function(){
+                        return domainObject.useCapability('composition')
+                        .then(function(composees){
+                            return self.$q.all(composees.map(function(composee){
+                                object.getCapability('composition').add(composee);
+                                return object;
+                            })).then(function(){return object});
+                        });
+                    });
+                }
+
+                return self.dialogService.getUserInput(
+                    wizard.getFormStructure(),
+                    wizard.getInitialFormValue()
+                ).then(persistResult, doNothing).then(composeObject);
+            }
 
             // Invoke any save behavior introduced by the editor capability;
             // this is introduced by EditableDomainObject which is
             // used to insulate underlying objects from changes made
             // during editing.
             function doSave() {
-                return domainObject.getCapability("editor").save();
+                //WARNING: HACK
+                //This is a new 'virtual panel' that has not been persisted
+                // yet.
+                if (domainObject.getModel().type === 'telemetry.panel' && !domainObject.getModel().persisted){
```

Comment:
My trick here would be to decorate the persistence capability of the created panels such that it popped up a dialog before persisting. Think this is nice because if future design changes mean that things get saved some other way, the rule "if we try to save an unsaved dynamically created object, show a dialog for it"

Actually, that rule might even be good enough to replace the Create dialog (since the create action is an instantiate plus a persist - we could hit that rule on the persist part, couldn't we?) But then we run into annoying little details like making sure an appropriate title appears in that dialog...

Alternately, if we really want this to be specific to Edit mode, I think the `editor` capability could be wrapped/decorated to pop this dialog up for telemetry panels specifically. (We also need to refactor out form creation for properties, such that it's easy to just conjure up a form section just by knowing a domain object's type.)